openfl/federated/task/runner_keras.py

```diff
@@ -189,13 +198,10 @@ def train_(self, batch_generator, metrics: list = None, **kwargs):
         #  defined) then the model must be recompiled.
         try:
             results = self.model.get_metrics_result()
+            if len(results) == 0:
+                results = self._initialize_metrics_result(kwargs)
         except ValueError:
-            if "batch_size" in kwargs:
-                batch_size = kwargs["batch_size"]
-            else:
-                batch_size = 1
-            # evaluation needed before metrics can be resolved
-            self.model.evaluate(self.data_loader.get_valid_loader(batch_size), verbose=1)
+            self._initialize_metrics_result(kwargs)
```

Comment:
`kwargs` can be replaced with `batch_size`, as per prev. comment

Reply:
Fixed in latest commit